### PR TITLE
Consume optimization data from bar & cleanup standard repo consumption

### LINKED ISSUE #35662: Make CoreFx pick up optimization data through darc rather than maestro v1
I've been trying to switch from using maestro v1 to darc for the `optimization.windows_nt-x64.IBC.CoreFx`

I've set up the darc subscription as
```
https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-optimization (.NET Core 3 Dev) ==> 'https://github.com/dotnet/corefx' ('master')
  - Id: dcaaa06b-98e3-4611-7eda-08d69dadeabb
  - Update Frequency: everyDay
  - Merge Policies:
    AllChecksSuccessful
      ignoreChecks =
                     [
                       "WIP",
                       "license/cla"
                     ]
  - Last Build: N/A
```

Note `.NET Core 3 Dev` channel.  This works for the PGO package that CoreCLR uses but not the IBC package.

Something is wrong with the publishing of dotnet-optimization such that the IBC package we depend on is not being updated.

/cc @hoyosjs @adiaaida

### PR DESCRIPTION
**Pull request by ericstj**
fixes #35662

### CODE REVIEW

**Comment by ericstj** [comment]
@hoyosjs helped me test this once.  dotnet-optimization still needs to iron out publishing but this side should be done.

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -91,5 +91,9 @@
       <Uri>https://github.com/dotnet/roslyn</Uri>
       <Sha>ec366687c8caead948e58704ad61ed644eb91b5b</Sha>
     </Dependency>
+    <Dependency Name="optimization.windows_nt-x64.IBC.CoreFx" Version="99.99.99-master-20190228.4">
```
Comment (safern) [inline eng/Version.Details.xml]: I recall seeing a linux package when you did the optimization work, is that another package?

Comment (safern) [reply eng/Version.Details.xml]: This guy: https://github.com/dotnet/corefx/pull/35675/files#diff-157c769233d1597c193c4aedfe05a488L28

Should we also add an entry for it?

Comment (safern) [reply eng/Version.Details.xml]: Oh sorry I see you're using the same version for both, disregard: https://github.com/dotnet/corefx/pull/35675/files#diff-bf787c0cad9f2c704edb4efb5c4d8400R14

Comment (ericstj) [reply eng/Version.Details.xml]: Currently it's set up to have the same version so I preserved what was there before.

**Comment by adiaaida** [comment]
We can also remove the extra myget source in the RestoreSources block in optimizationData.depproj (https://github.com/dotnet/corefx/blob/c8e44c3d027d0539e7f9091336cc48e59e7d03e8/external/optimizationData/optimizationData.depproj#L5). We are pushing to the standard dotnetfeed blob storage now, and would like to turn off the push to myget.

**Pull request merged**